### DISCUSSION TITLE
Proposal: Restructuring Types for Better Component Organization

### DISCUSSION THREAD

**Discussion by Fritsch-Tech** [gremo/react-directus#371] [Ideas]
#### Discussion/Idea

**Restructuring Types: Current Approach and Proposal**

Currently, all types are centralized in one file, `types.ts`. While this approach may serve its purpose, I'm curious if there's a specific reason for this choice or if it's your preferred way in general.

I would like to propose an alternative approach where we add the prop and return types directly in the file for the respective Component where they are used. This would localize the types, making it easier to manage and understand the type definitions for each Component.

Please share your thoughts and opinions on this proposal!

Thank you for participating in this discussion!

**Comment by gremo** [comment]
I'm not against this proposal (totally agree - if type is near the component, it's better to understand it).... but question is: how to manage shared types? Maybe now we haven't common types, but in the future you will end up having `types.ts` again...

**Reply by Fritsch-Tech** [reply]
For those common types, I would suggest creating a `commons` folder with individual type files inside. This approach would lead to a smaller number of files compared to the current `types` file and help in better organization and maintainability.
